synapseutils/copy.py

```diff
@@ -194,24 +195,31 @@ def _copyRecursive(syn, entity, destinationId, mapping=None, skipCopyAnnotations
     if not isinstance(ent, (Project, Folder, File, Link, Schema, Entity)):
         raise ValueError("Not able to copy this type of file")
 
-    if isinstance(ent, Project):
-        if not isinstance(syn.get(destinationId), Project):
-            raise ValueError("You must give a destinationId of a new project to copy projects")
-        copiedId = destinationId
-        entities = syn.getChildren(entity)
-        for i in entities:
-            mapping = _copyRecursive(syn, i['id'], destinationId, mapping=mapping,
-                                     skipCopyAnnotations=skipCopyAnnotations, **kwargs)
-    elif isinstance(ent, Folder):
-        copiedId = _copyFolder(syn, ent.id, destinationId, mapping=mapping, skipCopyAnnotations=skipCopyAnnotations,
-                               **kwargs)
-    elif isinstance(ent, File) and "file" not in excludeTypes:
-        copiedId = _copyFile(syn, ent.id, destinationId, version=version, updateExisting=updateExisting,
-                             setProvenance=setProvenance, skipCopyAnnotations=skipCopyAnnotations)
-    elif isinstance(ent, Link) and "link" not in excludeTypes:
-        copiedId = _copyLink(syn, ent.id, destinationId, updateExisting=updateExisting)
-    elif isinstance(ent, Schema) and "table" not in excludeTypes:
-        copiedId = _copyTable(syn, ent.id, destinationId, updateExisting=updateExisting)
+    profile_username = syn.getUserProfile()['userName']
+    permissions = syn.getPermissions(ent, profile_username)
+
+    access_requirements = syn.restGET('/entity/%s/accessRequirement' % ent.id)
+
+    #Do not copy anything with access requirements
+    if access_requirements['totalNumberOfResults'] == 0:
```

Comment:
this will not handle if an AR is added to a team, only directly to an entity or container.

Reply:
prefer early return so that the main chunk of the code is not forced to be extra indented 
```
move line 178 `copiedId = None` down here. No usage of that variable until we reach this point

if access_requirements['totalNumberOfResults'] == 0:
   print("%s not copied" % ent.id)
   return mapping

//rest of the if/elif logic
```

this also would allow you to remove the `if copiedId is not None` on line 224